### DISCUSSION TITLE
Mesh Index of a Given Finite Element Object

### DISCUSSION THREAD

**Discussion by FH-9** [mfem/mfem#4329] [Q&A] [answered]
Hi,

Given a `FiniteElement` object is it possible to infer its index in the mesh? I am writing a custom `AssembleElementVector` method that takes as an argument a `FiniteElement` object. However, I also need to access that element's neighbors for my application. I know how to do this if I am iterating over the elements in the mesh myself but not when I am just given a `FiniteElement` object. I would appreciate any pointers. Thank you very much.

**Comment by tzanio** [comment]
Do you mean `FiniteElement` or `Element`?

For the latter, you can consider encoding the element number as its attribute.

**Comment by pazner** [comment, marked as answer]
In `AssembleElementVector`, the given `ElementTransformation` has public member data `ElementTransformation::ElementNo`. This will correspond to the index of the given element. You can then use that information to find the neighbors.

**Comment by FH-9** [comment]
Hi @tzanio and @pazner,

Yes, I figured that out a little after posting this question :) Thanks nonetheless and apologies for a hasty question.
> In AssembleElementVector, the given ElementTransformation has public member data ElementTransformation::ElementNo. This will correspond to the index of the given element. You can then use that information to find the neighbors.
